### ISSUE TITLE
How to create a project in presidio?

### ISSUE THREAD

**Issue by teo-chenglim** [Microsoft/presidio#123]
**Describe the bug**
How to create a project in presidio? Trying hard to follow the tutorial
https://github.com/Microsoft/presidio/blob/master/docs/tutorial_service.md

**To Reproduce**
```bash
$ curl -X POST localhost:3000/api/v1/projects/1
Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.

$ curl -X POST localhost:3000/api/v1/projects/my-projects
Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.
```

**Expected behavior**
Before I can analyze a template.. I need to create a default project?
$ echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC333991", "analyzeTemplate":{"fields":[]}  }' | http <api-service-address>/api/v1/projects/<my-project>/analyze

**Screenshots**
N/A

**Additional context**
Just try to run the example from https://github.com/Microsoft/presidio/blob/master/docs/tutorial_service.md

**Comment by omri374** [comment]
Hi @teo-chenglim, thanks for your feedback. 
There is no need to set up a presidio project. In the next release we are planning on refining the documentation to avoid confusion.

If you're setting up a new template, just put an arbitrary project name and make sure to have the same project name in your API requests. If you call the API directly, just specify an arbitrary project name.
I will keep the issue open until the documentation is sorted out.

**Comment by teo-chenglim** [comment]
Hi,

thank you for the reply. Can you help me how to get this things work on my minikube? or docker?

```bash
$ echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC111921", "analyzeTemplate":{"fields":[]}  }' | http -F localhost:3000/api/v1/projects/1/analyze

http: error: ConnectionError: ('Connection aborted.', BadStatusLine('\x00\x00\x18\x04\x00\x00\x00\x00\x00\x00\x04\x00@\x00\x00\x00\x05\x00@\x00\x00\x00\x06\x00\x00 \x00þ\x03\x00\x00\x00\x01\x00\x00\x04\x08\x00\x00\x00\x00\x00\x00?\x00\x01\x00\x00\x08\x06\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',)) while doing POST request to URL: http://localhost:3000/api/v1/projects/1/analyze
$ echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC111921", "analyzeTemplate":{"fields":[]}  }' | http -F 192.168.1.44/api/v1/projects/1/analyze

http: error: SSLError: HTTPSConnectionPool(host='192.168.1.44', port=443): Max retries exceeded with url: /api/v1/projects/1/analyze (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),)) while doing POST request to URL: https://192.168.1.44/api/v1/projects/1/analyze
$ echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC111921", "analyzeTemplate":{"fields":[]}  }' | http -F https://192.168.1.44/api/v1/projects/1/analyze

http: error: SSLError: HTTPSConnectionPool(host='192.168.1.44', port=443): Max retries exceeded with url: /api/v1/projects/1/analyze (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),)) while doing POST request to URL: https://192.168.1.44/api/v1/projects/1/analyze
$ 
```

**Comment by teo-chenglim** [comment]
I have figure it out. Thank you for the tips tough.

```bash
$ echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC111921", "analyzeTemplate":{"fields":[]}  }' | http -F --verify=no  https://192.168.1.44/api/v1/projects/1/analyze
HTTP/1.1 200 OK
Connection: keep-alive
Content-Encoding: gzip
Content-Type: application/json; charset=utf-8
Date: Mon, 29 Apr 2019 08:14:42 GMT
Server: nginx/1.15.6
Transfer-Encoding: chunked
Vary: Accept-Encoding

[
    {
        "field": {
            "name": "PERSON"
        },
        "location": {
            "end": 10,
            "length": 10
        },
        "score": 0.85,
        "text": "John Smith"
    },
    {
        "field": {
            "name": "LOCATION"
        },
        "location": {
            "end": 28,
            "length": 8,
            "start": 20
        },
        "score": 0.85,
        "text": "New York"
    },
    {
        "field": {
            "name": "DATE_TIME"
        },
        "location": {
            "end": 46,
            "length": 9,
            "start": 37
        },
        "score": 0.85,
        "text": "yesterday"
    },
    {
        "field": {
            "name": "DATE_TIME"
        },
        "location": {
            "end": 54,
            "length": 7,
            "start": 47
        },
        "score": 0.85,
        "text": "morning"
    },
    {
        "field": {
            "name": "LOCATION"
        },
        "location": {
            "end": 65,
            "length": 7,
            "start": 58
        },
        "score": 0.85,
        "text": "Seattle"
    },
    {
        "field": {
            "name": "PHONE_NUMBER"
        },
        "location": {
            "end": 104,
            "length": 14,
            "start": 90
        },
        "score": 1,
        "text": "(212) 555-1234"
    },
    {
        "field": {
            "name": "US_DRIVER_LICENSE"
        },
        "location": {
            "end": 184,
            "length": 8,
            "start": 176
        },
        "score": 0.65,
        "text": "AC111921"
    }
]
```

**Comment by teo-chenglim** [comment]
by the way curl version also works. But still need jq for the color and the correct json header..

```bash
$ echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC111921", "analyzeTemplate":{"fields":[]}  }' | curl -k -s -H "Content-Type: application/json" https://192.168.1.44/api/v1/projects/1/analyze -d @- | jq
[
  {
    "text": "John Smith",
    "field": {
      "name": "PERSON"
    },
    "score": 0.85,
    "location": {
      "end": 10,
      "length": 10
    }
  },
  {
    "text": "New York",
    "field": {
      "name": "LOCATION"
    },
    "score": 0.85,
    "location": {
      "start": 20,
      "end": 28,
      "length": 8
    }
  },
  {
    "text": "yesterday",
    "field": {
      "name": "DATE_TIME"
    },
    "score": 0.85,
    "location": {
      "start": 37,
      "end": 46,
      "length": 9
    }
  },
  {
    "text": "morning",
    "field": {
      "name": "DATE_TIME"
    },
    "score": 0.85,
    "location": {
      "start": 47,
      "end": 54,
      "length": 7
    }
  },
  {
    "text": "Seattle",
    "field": {
      "name": "LOCATION"
    },
    "score": 0.85,
    "location": {
      "start": 58,
      "end": 65,
      "length": 7
    }
  },
  {
    "text": "(212) 555-1234",
    "field": {
      "name": "PHONE_NUMBER"
    },
    "score": 1,
    "location": {
      "start": 90,
      "end": 104,
      "length": 14
    }
  },
  {
    "text": "AC111921",
    "field": {
      "name": "US_DRIVER_LICENSE"
    },
    "score": 0.65,
    "location": {
      "start": 176,
      "end": 184,
      "length": 8
    }
  }
]
```

**Comment by omri374** [comment]
Reopening - Project configuration is misleading

**Comment by omri374** [comment]
Documentation was updated. Closing issue

**Comment by ariklevin** [comment]
Hi,

I have the same errors as you did and still didn't manage to run it successfully 

My docker compose looks like this: 
```  kryon-presidio-analyzer:
    image: mcr.microsoft.com/presidio-analyzer
    container_name: kryon-presidio-analyzer
    restart: unless-stopped  
    ports:
      - "3001:3001"  
    environment:
      GRPC_PORT: 3001
```
I run the following request:

```
echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC111921", "analyzeTemplate":{"fields":[]}  }' | http http://localhost:3000/api/v1/projects/1/analyze
```
and it results with 

```
http: error: ConnectionError: HTTPConnectionPool(host='localhost', port=3000): Max retries exceeded with url: /api/v1/projects/1/analyze (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000020C23DB8848>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it')) while doing a POST request to URL: http://localhost:3000/api/v1/projects/1/analyze
```

Any thoughts will be appreciated


> I have figure it out. Thank you for the tips tough.
> 
> ```shell
> $ echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC111921", "analyzeTemplate":{"fields":[]}  }' | http -F --verify=no  https://192.168.1.44/api/v1/projects/1/analyze
> HTTP/1.1 200 OK
> Connection: keep-alive
> Content-Encoding: gzip
> Content-Type: application/json; charset=utf-8
> Date: Mon, 29 Apr 2019 08:14:42 GMT
> Server: nginx/1.15.6
> Transfer-Encoding: chunked
> Vary: Accept-Encoding
> 
> [
>     {
>         "field": {
>             "name": "PERSON"
>         },
>         "location": {
>             "end": 10,
>             "length": 10
>         },
>         "score": 0.85,
>         "text": "John Smith"
>     },
>     {
>         "field": {
>             "name": "LOCATION"
>         },
>         "location": {
>             "end": 28,
>             "length": 8,
>             "start": 20
>         },
>         "score": 0.85,
>         "text": "New York"
>     },
>     {
>         "field": {
>             "name": "DATE_TIME"
>         },
>         "location": {
>             "end": 46,
>             "length": 9,
>             "start": 37
>         },
>         "score": 0.85,
>         "text": "yesterday"
>     },
>     {
>         "field": {
>             "name": "DATE_TIME"
>         },
>         "location": {
>             "end": 54,
>             "length": 7,
>             "start": 47
>         },
>         "score": 0.85,
>         "text": "morning"
>     },
>     {
>         "field": {
>             "name": "LOCATION"
>         },
>         "location": {
>             "end": 65,
>             "length": 7,
>             "start": 58
>         },
>         "score": 0.85,
>         "text": "Seattle"
>     },
>     {
>         "field": {
>             "name": "PHONE_NUMBER"
>         },
>         "location": {
>             "end": 104,
>             "length": 14,
>             "start": 90
>         },
>         "score": 1,
>         "text": "(212) 555-1234"
>     },
>     {
>         "field": {
>             "name": "US_DRIVER_LICENSE"
>         },
>         "location": {
>             "end": 184,
>             "length": 8,
>             "start": 176
>         },
>         "score": 0.65,
>         "text": "AC111921"
>     }
> ]
> ```

**Comment by SujithThazhathveedu** [comment]
I have deployed Presidio using the docker deployment
I see the 4 docker container running 
![image](https://user-images.githubusercontent.com/69573536/100480216-a14aaf80-3116-11eb-8d3e-78a65a85bdca.png)
i try to run this 
$ echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC111921", "analyzeTemplate":{"fields":[]}  }' | http -F --verify=
no  http://172.105.253.118/api/v1/projects/1/analyze

and i get this 
 command not found

http: error: ConnectionError: HTTPConnectionPool(host='172.105.253.118', port=80): Max retries exceeded with url: /api/v1/projects/1/analyze (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fefb6735128>: Failed to establish a new connection: [Errno 111] Connection refused',)) while doing POST request to URL: http://172.105.253.118/api/v1/projects/1/analyze

**Comment by nikhil-neu** [comment]
I have deployed Presidio using the docker deployment
I see the 4 presidio docker containers running

i try to run this
$ echo -n '{"text":"John Smith lives in New York. We met yesterday morning in Seattle. I called him before on (212) 555-1234 to verify the appointment. He also told me that his drivers license is AC111921", "analyzeTemplate":{"fields":[]} }' | http -F --verify=
no http://172.105.253.118/api/v1/projects/1/analyze

and i get this
command not found

http: error: ConnectionError: HTTPConnectionPool(host='172.105.253.118', port=80): Max retries exceeded with url: /api/v1/projects/1/analyze (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fefb6735128>: Failed to establish a new connection: [Errno 111] Connection refused',)) while doing POST request to URL: http://172.105.253.118/api/v1/projects/1/analyze
